storagenode/gracefulexit/worker.go

```diff
@@ -91,10 +93,13 @@ func (worker *Worker) Run(ctx context.Context, done func()) (err error) {
 		case *pb.SatelliteMessage_NotReady:
 			break // wait until next worker execution
 		case *pb.SatelliteMessage_TransferPiece:
-			err = worker.transferPiece(ctx, msg.TransferPiece, c)
-			if err != nil {
-				continue
-			}
+			transferPieceMsg := msg.TransferPiece
+			worker.limiter.Go(ctx, func() {
+				err = worker.transferPiece(ctx, transferPieceMsg, c)
+				if err != nil {
+					worker.log.Error("failed to transfer piece.", zap.Stringer("Satellite ID", worker.satelliteID), zap.Error(errs.Wrap(err)))
```

Comment:
Maybe we should remove all logs from inside `worker.transferPiece` and just have a single log here

Reply:
I think that would make sense